src/main/scala/com/bazaarvoice/sswf/model/history/HistoryFactory.scala

```diff
@@ -88,15 +89,16 @@ object HistoryFactory {
             val id: String = attributes.getActivityId
             val (stepInput, _) = unpackInput(inputParser)(attributes.getInput)
             val scheduledStep = DefinedStep(enumFromName(id), stepInput)
-            if (currentStep != scheduledStep) {
+            if (currentStep == null) {
```

Comment:
yeah, if we're scheduling subsequent step executions after IN_PROGRESS, for example.